### PR TITLE
expression: Change escape character from "$" to "\" for `regexp_replace`

### LINKED ISSUE #44684: Change escape character from "$" to "\" for `regexp_replace`

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: close #44684

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: expression/builtin_regexp_test.go [hunk 1]
```diff
@@ -948,6 +948,9 @@ func TestRegexpReplace(t *testing.T) {
 		{"abc abd abe", "ab.", "cz", "cz cz cz", "0x637A20637A20637A", nil},
 		{"你好 好的", "好", "逸", "你逸 逸的", "0xE4BDA0E980B820E980B8E79A84", nil},
 		{"", "^$", "123", "123", "0x313233", nil},
+		{"stackoverflow", "(.{5})(.*)", "\\\\+\\2+\\1+\\2+\\1\\", "\\+overflow+stack+overflow+stack", "0x5C2B6F766572666C6F772B737461636B2B6F766572666C6F772B737461636B", nil},
+		{"fooabcdefghij fooABCDEFGHIJ", "foo(.)(.)(.)(.)(.)(.)(.)(.)(.)(.)", "\\\\\\9\\\\\\8-\\7\\\\\\6-\\5\\\\\\4-\\3\\\\\\2-\\1\\\\", "\\i\\h-g\\f-e\\d-c\\b-a\\ \\I\\H-G\\F-E\\D-C\\B-A\\", "0x5C395C382D375C362D355C342D335C322D315C205C395C382D375C362D355C342D335C322D315C", nil},
```
Comment (gengliqi) [inline expression/builtin_regexp_test.go:952]: Using \` instead of " can make this string a raw string literal which can reduce `\`.

Comment (xzhangxian1008) [reply expression/builtin_regexp_test.go:952]: > Using ` instead of " can make this string a raw string literal which can reduce `\`.

okk

**Review by gengliqi — APPROVED** [review]

**Review by windtalker — APPROVED** [review]
LGTM

**Review by guo-shaoge — APPROVED** [review]

**Pull request closed without merging**